umake/ui/cli/__init__.py

```diff
@@ -157,6 +157,60 @@ def mangle_args_for_default_framework(args):
     return result_args
 
 
+def list_frameworks(listing, detailled=True):
+    if listing == "all" or listing == "available":
+        # Sort the categories to prevent a random list at each new program execution
+        for category in sorted(BaseCategory.categories.values(), key=lambda cat: cat.prog_name):
+            # The category must have at least one framework
+            if len(category.frameworks) == 0:
+                continue
+
+            if detailled:
+                print("{}: {}".format(category.prog_name, category.description))
+            else:
+                print("{}:".format(category.prog_name))
+
+            # Sort the frameworks to prevent a random list at each new program execution
+            for framework in sorted(category.frameworks.values(), key=lambda fram: fram.prog_name):
+                # Framework doesn't be removal
+                if framework.only_for_removal:
+                    continue
+
+                if listing == "available":
+                    if not framework.is_installable:
+                        continue
+
+                if detailled:
+                    # Attribute 'description' of 'framework' contains if the category is default
+                    print("  {}: {} ".format(framework.prog_name, framework.description), end="")
```

Comment:
I didn't know that. I think I'm not accustomed to translate my program. The character `_` represents the `gettext` library, is that right? 
Thank you for this suggestion, I will apply it now!

> Same, you should use "\n" for line breaks in the previous sentence you are using, you should tend to avoid print() alone (this may be needed in some occasion, like when you want to explicitely break a line between categories).
I agree with you, I remembered when I wrote this line I was not sure about a print() which print nothing ^^